test_package/test-pull/CMakeLists.txt

```diff
@@ -8,6 +8,6 @@ set_target_properties(sample_server
 )
 
 target_link_libraries(sample_server
-  PRIVATE
-    CONAN_PKG::prometheus-cpp
```

Comment:
I thought it is the prefered way. Making explicit what that library depends on.

Reply:
Yes, but something is not working. But of course, it's related to Conan generator, otherwise it should work when using target.